{
  try {
    executor.execute(new Runnable(){
      @Override public void run(){
        try {
          Object resolvedResponse=resolve(response);
          if (resolvedResponse == null || !(resolvedResponse instanceof Throwable)) {
            callback.onResponse((E)resolvedResponse);
          }
 else {
            Throwable error=(Throwable)resolvedResponse;
            if (error instanceof ExecutionException) {
              error=error.getCause();
            }
            callback.onFailure(error);
          }
        }
 catch (        Throwable cause) {
          invocation.logger.severe("Failed asynchronous execution of execution callback: " + callback + "for call "+ invocation,cause);
        }
      }
    }
);
  }
 catch (  RejectedExecutionException e) {
    invocation.logger.warning("Execution of callback: " + callback + " is rejected!",e);
  }
}
